snapcraft/tests/test_plugin_catkin.py

```diff
@@ -44,107 +45,255 @@ class props:
-                               "package.xml"), 'w') as f:
-            f.write("""<?xml version="1.0"?>
-                      <package>
-                      </package>""")
+        self.rosdep_mock.return_value.get_dependencies.side_effect = get
 
         plugin.pull()
 
-        self.assertTrue('my_package' in plugin.package_local_deps,
-                        'Expected "my_package" to be in the dependencies')
-        self.assertEqual(plugin.package_local_deps['my_package'],
-                         {'package_2'},
-                         'Expected "my_package" to depend upon "package_2"')
-        self.assertFalse(self.ubuntu_mock.called,
-                         "Ubuntu packages were unexpectedly pulled down")
+        # Verify that rosdep was setup
+        self.rosdep_mock.assert_has_calls([
+            mock.call(self.properties.rosdistro,
+                      os.path.join(plugin.sourcedir, 'src'),
+                      os.path.join(plugin.partdir, 'rosdep'),
+                      plugin.PLUGIN_STAGE_SOURCES),
+            mock.call().setup()])
+
+        # Verify that rosdep was called to obtain dependencies
+        self.assertEqual(
+            self.rosdep_mock.return_value.get_dependencies.call_count, 2)
+        self.rosdep_mock.return_value.get_dependencies.assert_has_calls([
+            mock.call('my_package'),
+            mock.call('package_2')], any_order=True)
+
+        # Verify that rosdep was NOT called to resolve dependencies
+        self.rosdep_mock.return_value.resolve_dependency.assert_not_called()
+
+        # Verify that the dependency tree is what we expect
+        self.assertEqual(plugin.package_local_deps, {
+            'my_package': {'package_2'},
+            'package_2': set(),
+        })
+
+        # Verify that no .deb packages were installed
+        self.assertTrue(mock.call().unpack(plugin.installdir) not in
+                        self.ubuntu_mock.mock_calls)
+
+    def test_pull_missing_local_dependency(self):
+        plugin = catkin.CatkinPlugin('test-part', self.properties)
+        os.makedirs(os.path.join(plugin.sourcedir, 'src'))
 
-    def test_log_warning_when_unable_to_find_a_catkin_package(self):
-        fake_logger = fixtures.FakeLogger()
-        self.useFixture(fake_logger)
+        # Fake rosdep to setup the following dependecy/resolver tree:
+        # my_package:
+        #   - package_2 = local dependency
+        #                 (which isn't included in the catkin_packages)
+        def get(package_name):
+            if package_name == 'my_package':
+                return ['package_2']
 
-        plugin = catkin.CatkinPlugin('test-part', self.properties)
+            return ''
+
+        self.rosdep_mock.return_value.get_dependencies.side_effect = get
+        self.rosdep_mock.return_value.resolve_dependency.return_value = None
+
+        with self.assertRaises(RuntimeError) as raised:
+            plugin.pull()
 
+        self.assertEqual(raised.exception.args[0],
```

Comment:
Ah, hey @sergiusens I missed your comment here. I feel like the former is better-looking, but I'm happy doing either.